code/modules/mob/living/carbon/human/update_icons.dm

```diff
@@ -354,49 +354,60 @@ GLOBAL_LIST_EMPTY(damage_icon_parts)
 		else
 			//warning("Invalid ha_style for [species.name]: [ha_style]")
 
-
-
-//HAIR OVERLAY
+/**
+  * Generates overlays for the hair layer.
+  */
 /mob/living/carbon/human/proc/update_hair()
```

Comment:
Completely rewrote this proc to not use icon ops, in favour of overlays. Icon ops are generally expensive and disk dependent. This should serve as the basis for a complete update icons refactor later.